girder/utility/server.py

```diff
@@ -64,13 +65,10 @@ def configureServer(test=False, plugins=None, curConfig=None):
     mimetypes.add_type('application/font-woff', '.woff')
 
     curConfig.update(appconf)
+    curConfig['server']['mode'] = mode
```

Comment:
I realized there is a subtle change here that could be the source of future bugs. The question is: "what is the source of truth for the default mode?":
1. The file `girder/conf/girder.dist.cfg`
2. The default parameter of `configureServer` of `girder/utility/server.py`
3. The default parameter of `setup` of `girder/utility/server.py`
4. The default argument value in `girder/cli/serve.py`
5. The default argument value in `girder/cli/build.py`

Put anther way, if we wanted to change the default server mode back to "development", and we didn't change all 5 of these places together, inconsistencies and bugs could arise.

So, I'd propose:
* make the file `girder/conf/girder.dist.cfg` the source of truth
* set the default function arguments to be `mode=None`, and only override values (like `curConfig['server']['mode'] = mode`) if mode is set / truthy
* make the default argument value from the CLIs be `None` / unset

Reply:
I agree. This also fixes the issue that I was having around not reading the config file properly.